bigtable-client-core-parent/bigtable-client-core/src/test/java/com/google/cloud/bigtable/config/TestUserAgent.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2019 Google LLC. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.bigtable.config;
+
+import com.google.bigtable.v2.BigtableGrpc;
+import com.google.bigtable.v2.MutateRowRequest;
+import com.google.bigtable.v2.MutateRowResponse;
+import com.google.bigtable.v2.ReadRowsRequest;
+import com.google.bigtable.v2.ReadRowsResponse;
+import com.google.cloud.bigtable.data.v2.BigtableDataClient;
+import com.google.cloud.bigtable.data.v2.BigtableDataSettings;
+import io.grpc.ForwardingServerCall;
+import io.grpc.Metadata;
+import io.grpc.Server;
+import io.grpc.ServerBuilder;
+import io.grpc.ServerCall;
+import io.grpc.ServerCallHandler;
+import io.grpc.ServerInterceptor;
+import io.grpc.ServerInterceptors;
+import io.grpc.stub.StreamObserver;
+import java.net.ServerSocket;
+
+import java.util.regex.Pattern;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import sun.jvm.hotspot.utilities.AssertionFailure;
+
+public class TestUserAgent {
+  private static final Logger logger = new Logger(TestUserAgent.class);
+
+  private static final String TEST_PROJECT_ID = "Project-Id";
+  private static final String TEST_INSTANCE_ID = "instance";
+  private static final String TEST_USER_AGENT = "sampleUserAgent";
+  private static final Pattern EXPECTED_HEADER_PATTERN =
+      Pattern.compile(BigtableVersionInfo.CORE_USER_AGENT + "," + TEST_USER_AGENT + ".*");
+
+  private BigtableDataClient dataClient;
+  private Server server;
+
+  @Before
+  public void setUp() throws Exception {
+    //After fetching available port, closing the ServerSocket.
+    ServerSocket serverSocket = new ServerSocket(0);
+    final int availablePort = serverSocket.getLocalPort();
+    serverSocket.close();
+
+    //Creating Server with ServerInterceptor, To fetch user-agent header.
```

Comment:
"Creating Server with ServerInterceptor, in order to fetch user-agent header." if you want to continue using the "ing" form. However, since this is a smaller file, it would be best to reword like this: 
"Creates Server with ServerInterceptor, in order to fetch user-agent header."

Reply:
Thanks for the suggestion. I will try to minimize the use ing from now on..